inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationConfigurationResolver.java

```diff
@@ -0,0 +1,118 @@
+package rocks.inspectit.oce.core.instrumentation.config;
+
+import lombok.Getter;
+import lombok.val;
+import net.bytebuddy.description.type.TypeDescription;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.oce.core.config.InspectitConfigChangedEvent;
+import rocks.inspectit.oce.core.config.InspectitEnvironment;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.instrumentation.AsyncClassTransformer;
+import rocks.inspectit.oce.core.instrumentation.special.SpecialSensor;
+
+import javax.annotation.PostConstruct;
+import java.lang.instrument.Instrumentation;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is responsible for deriving the {@link InstrumentationConfiguration} from teh {@link InstrumentationSettings}.
+ */
+@Service
+public class InstrumentationConfigurationResolver {
+
+    private static final ClassLoader INSPECTIT_CLASSLOADER = AsyncClassTransformer.class.getClassLoader();
+
+    @Autowired
+    private InspectitEnvironment env;
+
+    @Autowired
+    private ApplicationContext ctx;
+
+    @Autowired
+    private Instrumentation instrumentation;
+
+    @Autowired
+    private List<SpecialSensor> specialSensors;
+
+    /**
+     * Holds the currently active instrumentation configuration.
+     */
+    @Getter
+    private InstrumentationConfiguration currentConfig;
+
+    @PostConstruct
+    private void init() {
+        updateConfiguration(env.getCurrentConfig().getInstrumentation());
+    }
+
+    /**
+     * Builds the {@link ClassInstrumentationConfiguration} based on the currently active global instrumentation configuration
+     * for the given class.
+     *
+     * @param clazz the class for which the configuration shal lbe queried
+     * @return the configuration or {@link ClassInstrumentationConfiguration#NO_INSTRUMENTATION} if this class should not be instrumented
+     */
+    public ClassInstrumentationConfiguration getClassInstrumentationConfiguration(Class<?> clazz) {
+        val config = currentConfig;
+        if (isIgnoredClass(clazz, config)) {
+            return ClassInstrumentationConfiguration.NO_INSTRUMENTATION;
+
+        } else {
+            TypeDescription description = TypeDescription.ForLoadedType.of(clazz);
+            Set<SpecialSensor> activeSensors = specialSensors.stream()
+                    .filter(s -> s.shouldInstrument(description, config))
+                    .collect(Collectors.toSet());
+            return new ClassInstrumentationConfiguration(activeSensors, config);
```

Comment:
I have a feeling that for not ignored classes we are actually always creating the `ClassInstrumentationConfiguration` instance with an empty set of sensors.. this way we are always trying to instrument the class imo, just never actually change the byte code as a loop in the transformer is skipped.. We should return the `NO_INSTRUMENTATION` is there is no active sensor!

Reply:
Solved by discussion: in case the class does not have any active sensor, a configuration is returned which "isSameAs" the NO_INSTRUMENTATION configuration.